### PR TITLE
Redux saga config

### PR DESCRIPTION
**Pull request by aczajczynska**
store, rootSaga, rootReducer, navigation types, sample counter,

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,13 +1,16 @@
...
 import { createStackNavigator } from 'react-navigation-stack';
+import { Provider } from 'react-redux';
 import { IntlProvider } from 'react-intl';
 import { ThemeProvider } from 'styled-components';
 import { defaultTheme } from '@styles/themes/defaultTheme';
 import { AppWrapper } from './app/styles/components/containers';
 
 import { navigationStructure } from './app/modules';
 
+import { rootStore } from './app/store/index';
```
Comment (adamkosmala) [inline src/App.tsx]: it should be `./app/store/`- no need to point to index file

File: src/app/modules/MainScreen/MainScreen.tsx [hunk 2]
```diff
@@ -1,29 +1,61 @@
 import React, { PureComponent } from 'react';
-import { Button } from 'react-native';
+// import { compose, bindActionCreators } from 'redux';
```
Comment (adamkosmala) [inline src/app/modules/MainScreen/MainScreen.tsx]: please remove all commented out code

File: src/app/modules/MainScreen/MainScreen.tsx [hunk 3]
```diff
@@ -1,29 +1,61 @@
...
-import { MainWrapper, MainTitle } from './styles';
+// import { injectIntl, WrappedComponentProps } from 'react-intl';
+import { WrappedComponentProps } from 'react-intl';
+
+import { MainWrapper, MainTitle } from './components/Main.styles';
 import Modules from '../index';
 
-type MainScreenProps = WrappedComponentProps & NavigationStackScreenProps;
+interface OwnProps {
+  decrementValue: any;
```
Comment (adamkosmala) [inline src/app/modules/MainScreen/MainScreen.tsx]: please add proper typings

File: src/app/modules/MainScreen/index.tsx [hunk 4]
```diff
@@ -1,7 +1,39 @@
...
+  incrementValue,
+  decrementValue,
+  incrementAsyncValue,
+} from './store/actions';
+
 import MainScreen from './MainScreen';
 
-export default {
-  module: MainScreen,
+const mapStateToProps = (state: any): any => {
```
Comment (adamkosmala) [inline src/app/modules/MainScreen/index.tsx]: please add proper typings

File: src/app/modules/MainScreen/index.tsx [hunk 5]
```diff
@@ -1,7 +1,39 @@
...
+  return {
+    value: state.ageValue.value,
+  };
+};
+
+const mapDispatchToProps = (dispatch: any): any => {
```
Comment (adamkosmala) [inline src/app/modules/MainScreen/index.tsx]: please add proper typings (there's `Dispatch` type available in Redux)

File: src/app/modules/MainScreen/store/actions/index.ts [hunk 6]
```diff
@@ -0,0 +1,27 @@
+import { ReturnType } from '../../../../namespace';
```
Comment (adamkosmala) [inline src/app/modules/MainScreen/store/actions/index.ts]: please use aliases

File: src/app/modules/MainScreen/index.tsx [hunk 7]
```diff
@@ -1,7 +1,39 @@
...
+    {
+      incrementValue,
+      decrementValue,
+      incrementAsyncValue,
+    },
+    dispatch,
+  );
+};
+
+const MainScreenComposed = compose(
```
Comment (adamkosmala) [inline src/app/modules/MainScreen/index.tsx]: what about having something like this:

```
import { NavigationScreenComponent } from 'react-navigation';
import { NavigationStackOptions } from 'react-navigation-stack';

....

const MainScreenComposed = compose<NavigationScreenComponent<any, any>>(
  injectIntl,
  connect(mapStateToProps, mapDispatchToProps),
)(MainScreen);

interface ScreenPropsConfig {
  options: NavigationStackOptions;
  module: NavigationScreenComponent<any, any>;
  name: string;
}

export const MainScreenModule: ScreenPropsConfig = {
  module: MainScreenComposed,
  name: 'MainScreen',
  options: {
    headerLeft: null,
    headerTitle: null,
    headerRight: null,
  },
};
```

`ScreenPropsConfig` interface would be a common type used for all modules.

@grzegorzkruk @lukaszblasz ?

File: package.json [hunk 8]
```diff
@@ -22,7 +22,9 @@
     "@types/react": "16.9.15",
     "@types/react-intl": "3.0.0",
```
Comment (lukaszblasz) [inline package.json]: Please refactor this, things like redux-logger, or typescript shoudl be in devdepedencies. Also all things related to typescript or used just for dev mode should be placed in dev dependecies.

File: src/App.tsx [hunk 9]
```diff
@@ -1,13 +1,16 @@
 import React from 'react';
 import { createAppContainer } from 'react-navigation';
 import { createStackNavigator } from 'react-navigation-stack';
+import { Provider } from 'react-redux';
 import { IntlProvider } from 'react-intl';
 import { ThemeProvider } from 'styled-components';
 import { defaultTheme } from '@styles/themes/defaultTheme';
 import { AppWrapper } from './app/styles/components/containers';
```
Comment (adamkosmala) [inline src/App.tsx]: can we use aliases everywhere?

File: src/app/modules/MainScreen/index.tsx (hunk 7 above)
Comment (lukaszblasz) [reply src/app/modules/MainScreen/index.tsx]: Im fine with that, but it shoudl not be placed there, So ScreenPropsConfig should be put in ru domain/common typing and imported here and also in every module in application.

File: src/app/modules/MainScreen/store/saga/index.ts [hunk 10]
```diff
@@ -0,0 +1,15 @@
+import { put, takeLatest, all, fork, delay } from 'redux-saga/effects';
+import { MainScreenTypes } from '../actions/index';
+
+function* incrementAync() {
+  yield delay(1000);
```
Comment (lukaszblasz) [inline src/app/modules/MainScreen/store/saga/index.ts]: number values as CONST and named properly.

File: src/app/modules/MainScreen/index.tsx [hunk 11]
```diff
@@ -10,29 +12,38 @@ import {
...
+  incrementValue: () => void;
+  decrementValue: () => void;
+  incrementAsyncValue: () => void;
+}
+
+export interface StateProps {
+  value: number;
+}
+
+const mapStateToProps = (state: any): StateProps => {
```
Comment (adamkosmala) [inline src/app/modules/MainScreen/index.tsx]: still missing type for state

File: src/app/modules/MainScreen/index.tsx [hunk 12]
```diff
@@ -10,29 +12,38 @@ import {
...
     },
     dispatch,
   );
-};
 
-const MainScreenComposed = compose(
+const MainScreenComposed = compose<NavigationScreenComponent<any, any>>(
   injectIntl,
-  connect(mapStateToProps, mapDispatchToProps),
+  connect<StateProps, DispatchProps>(mapStateToProps, mapDispatchToProps),
```
Comment (adamkosmala) [inline src/app/modules/MainScreen/index.tsx]: :+1:

File: src/app/modules/index.tsx [hunk 13]
```diff
@@ -1,9 +1,7 @@
 import { MainScreenModule } from './MainScreen';
 import { ProfileScreenModule } from './ProfileScreen';
 
-import { NavigationStructureTypes } from '../shared/types/navigation';
-
-export const navigationStructure: NavigationStructureTypes = {
+export const navigationStructure = {
```
Comment (adamkosmala) [inline src/app/modules/index.tsx]: let's have something like that as well:

```
interface NavigationStructureProps {
  [key: string]: {
    screen: NavigationScreenComponent<any, any>;
    navigationOptions: NavigationStackOptions;
  };
}

export const navigationStructure: NavigationStructureProps =  { ...}

```

While `NavigationStructureProps` would be placed in `namespace` folder

File: src/app/modules/MainScreen/index.tsx [hunk 14]
```diff
@@ -10,29 +12,38 @@ import {
...
+const mapStateToProps = (state: any): StateProps => {
   return {
     value: state.ageValue.value,
   };
 };
 
-const mapDispatchToProps = (dispatch: any): any => {
-  return bindActionCreators(
+const mapDispatchToProps = (dispatch: Dispatch): DispatchProps =>
+  bindActionCreators(
```
Comment (adamkosmala) [inline src/app/modules/MainScreen/index.tsx]: Personally I'm not a fan of this syntax, but it's fine

File: src/app/modules/MainScreen/store/reducers/index.ts [hunk 15]
```diff
@@ -0,0 +1,23 @@
+import { MainScreenTypes, MainScreenActions } from '../actions';
+
+const inistalState = {
```
Comment (adamkosmala) [inline src/app/modules/MainScreen/store/reducers/index.ts]: spelling mistake- it should be `initialState` :)

**Pull request merged**